src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AutoValue
+public abstract class JavaDocComment implements Comment {
+
+  public abstract ImmutableList<String> comments();
+
+  public abstract ImmutableList<String> params();
+
+  public abstract String deprecated();
+
+  // Private.
+  abstract String throwType();
+
+  // Private.
+  abstract String throwDescription();
+
+  public static Builder builder() {
+    return new AutoValue_JavaDocComment.Builder().setDeprecated("").setThrows("", "");
+  }
+
+  @Override
+  public String comment() {
+    // TODO(xiaozhenliu): call comment escaper here.
+    List<String> comment = comments().stream().collect(Collectors.toList());
+    // @param, @throws and @deprecated should always get printed at the end.
+    comment.addAll(params().stream().collect(Collectors.toList()));
+    if (!throwType().isEmpty()) {
```

Comment:
Shouldn't this occur in `Builder.build()`?

Reply:
not sure why it should occur in `Builder.build()`, can you explain more?  For my understanding `Builder()` collects all the parts of a javadocComment components and in override `comment()`, we group them and return the consolidated comment.

Reply:
The `comment()` method should return the entire comment contents (to be consistent with the other comment classes), and it seems that `build()` currently collects and formats the entire string. Ergo, `build()` should also "write" the `@deprecated`, `@throws`, and `@param` tags.

Reply:
This sounds reasonable to me! Moved the `@deprecated` `@throws` `@params` components in `build()` method, and have a private `setComment()` to set the formatted, complete, consolidated comment to the `comment()`.